### PR TITLE
WIP: test a few CLI changes

### PR DESCRIPTION
**Pull request by brb**

### CODE REVIEW

**Comment by jschwinger233** [comment]
https://github.com/cilium/cilium/actions/runs/6097405526/job/16544976926
The failure `connectivity test failed: timeout reached waiting for deployment cilium-test/test-conn-disrupt-client to become ready (last error: only 3 of 5 replicas are available)` could be caused by previous connectivity test with `--flush-ct`? According to output of `kubectl get po`, the disrupt-conn-client pods crashed at the end of last connectivity test, which is the moment to flush ct.

**Comment by brb** [comment]
> According to output of kubectl get po, the disrupt-conn-client pods crashed at the end of last connectivity test, which was the moment to flush ct.

@jschwinger233 Maybe we should improve the CLI tests by properly removing the disrupt-conn-{client,server} pods?

**Comment by jschwinger233** [comment]
@brb I am just aware of `--force-deploy`which forces to re-deploy test pods including conn-disruption pods. How about enabling this flag for ipsec-upgrade-test and ipsec-key-rotation-test(ipsec-ci-e2e)?

File: .github/workflows/tests-ipsec-upgrade.yaml [hunk 1]
```diff
@@ -261,9 +261,8 @@ jobs:
         uses: cilium/cilium/.github/actions/conn-disrupt-test@d86f33b098a75f3d7d2dfd3ee6fa1ea033892de4
         with:
           job-name: ipsec-upgrade-${{ matrix.name }}
-          # Disable no-missed-tail-calls due to https://github.com/cilium/cilium/issues/26739
           # Disable check-log-errors due to https://github.com/cilium/cilium-cli/issues/1858
-          extra-connectivity-test-flags: --test '!no-missed-tail-calls,!check-log-errors'
+          extra-connectivity-test-flags: --test '!check-log-errors'
```
Comment (jschwinger233) [inline .github/workflows/tests-ipsec-upgrade.yaml:265]:
```suggestion
          extra-connectivity-test-flags: --test '!check-log-errors' --force-deploy
```

File: .github/workflows/tests-ipsec-upgrade.yaml [hunk 2]
```diff
@@ -278,9 +277,8 @@ jobs:
         uses: cilium/cilium/.github/actions/conn-disrupt-test@d86f33b098a75f3d7d2dfd3ee6fa1ea033892de4
         with:
           job-name: ipsec-downgrade-${{ matrix.name }}
-          # Disable no-missed-tail-calls due to https://github.com/cilium/cilium/issues/26739
           # Disable check-log-errors due to https://github.com/cilium/cilium-cli/issues/1858
-          extra-connectivity-test-flags: --test '!no-missed-tail-calls,!check-log-errors'
+          extra-connectivity-test-flags: --test '!check-log-errors'
```
Comment (jschwinger233) [inline .github/workflows/tests-ipsec-upgrade.yaml:281]: Same as above.

**Comment by brb** [comment]
I was thinking to delete just the test-conn-disrupt pods to reduce the runtime of the upgrade tests, e.g., `cilium connectivity --include-conn-disrupt-test --del-conn-disrupt-test-pods` which after running the tests would remove the pods.

File: .github/workflows/tests-ipsec-upgrade.yaml (hunk 1 above)
Comment (brb) [reply .github/workflows/tests-ipsec-upgrade.yaml:265]: Let me try this quick fix :+1:

**Comment by brb** [comment]
@jschwinger233 Trying out https://github.com/cilium/cilium-cli/pull/1952/files

**Comment by brb** [comment]
@jschwinger233 https://github.com/cilium/cilium/actions/runs/6110635519/job/16584099256 - It's green! Let me try again.

**Comment by jschwinger233** [comment]
https://github.com/cilium/cilium/actions/runs/6113949121 Still green, let's try one more time

**Comment by brb** [comment]
@jschwinger233 :sob: 
>   🟥 Pod test-conn-disrupt-client-cc557fcf9-j5zjz flow was interrupted (restart count does not match 1 != 2)

**Comment by brb** [comment]
@jschwinger233 Let's try adjusting the sleep of test-conn-disrupt?

**Comment by jschwinger233** [comment]
> @jschwinger233 Let's try adjusting the sleep of test-conn-disrupt?

https://github.com/cilium/test-connection-disruption/pulls 

Is this okay?
